diff -Naurp mdbtools/src/gmdb2/main2.c mdbtools.oden/src/gmdb2/main2.c
--- mdbtools/src/gmdb2/main2.c	2005-01-15 06:02:12.000000000 +0100
+++ mdbtools.oden/src/gmdb2/main2.c	2008-12-21 20:26:53.000000000 +0100
@@ -90,7 +90,7 @@ gmdb_help_cb(GtkWidget *button, gpointer
 
 	gnome_help_display("gmdb.xml", NULL, &error);
 	if (error != NULL) {
-		g_warning (error->message);
+		g_warning ("%s", error->message);
 		g_error_free (error);
 	}
 
diff -Naurp mdbtools/src/gmdb2/prefs.c mdbtools.oden/src/gmdb2/prefs.c
--- mdbtools/src/gmdb2/prefs.c	2005-01-15 06:02:12.000000000 +0100
+++ mdbtools.oden/src/gmdb2/prefs.c	2008-12-21 20:26:53.000000000 +0100
@@ -42,7 +42,7 @@ gmdb_prefs_help_cb(GtkWidget *w, gpointe
 
 	gnome_help_display("gmdb.xml", "gmdb-prefs", &error);
 	if (error != NULL) {
-		g_warning (error->message);
+		g_warning ("%s", error->message);
 		g_error_free (error);
 	}
 }
diff -Naurp mdbtools/src/gmdb2/schema.c mdbtools.oden/src/gmdb2/schema.c
--- mdbtools/src/gmdb2/schema.c	2005-01-15 06:02:12.000000000 +0100
+++ mdbtools.oden/src/gmdb2/schema.c	2008-12-21 20:26:53.000000000 +0100
@@ -168,7 +168,7 @@ gmdb_schema_help_cb(GtkWidget *w, gpoint
 
 	gnome_help_display("gmdb.xml", "gmdb-schema", &error);
 	if (error != NULL) {
-		g_warning (error->message);
+		g_warning ("%s", error->message);
 		g_error_free (error);
 	}
 }
diff -Naurp mdbtools/src/gmdb2/sql.c mdbtools.oden/src/gmdb2/sql.c
--- mdbtools/src/gmdb2/sql.c	2005-01-15 06:02:12.000000000 +0100
+++ mdbtools.oden/src/gmdb2/sql.c	2008-12-21 20:26:53.000000000 +0100
@@ -104,7 +104,7 @@ gmdb_sql_write_rslt_cb(GtkWidget *w, Gla
 	if (need_headers)  {
 		while (col = g_list_nth_data(glist, i)) {
 			gchar *title;
-			if (i>0) fprintf(outfile,delimiter);
+			if (i>0) fprintf(outfile, "%s", delimiter);
 			title = g_strdup(gtk_tree_view_column_get_title(col));
 			gmdb_print_quote(outfile, need_quote, quotechar,
 				delimiter, title);
@@ -114,7 +114,7 @@ gmdb_sql_write_rslt_cb(GtkWidget *w, Gla
 			g_free(title);
 			i++;
 		}
-		fprintf(outfile,lineterm);
+		fprintf(outfile, "%s", lineterm);
 		g_list_free(glist);
 	}
 
@@ -126,7 +126,7 @@ gmdb_sql_write_rslt_cb(GtkWidget *w, Gla
 		rows++;
 		n_columns = gtk_tree_model_get_n_columns(GTK_TREE_MODEL(store));
 		for (i=0; i < n_columns; i++) {
-			if (i>0) fprintf(outfile,delimiter);
+			if (i>0) fprintf(outfile, "%s", delimiter);
 			gtk_tree_model_get_value(GTK_TREE_MODEL(store), 
 					&iter, i, &value);
 			str = (gchar *) g_value_get_string(&value);
@@ -135,7 +135,7 @@ gmdb_sql_write_rslt_cb(GtkWidget *w, Gla
 			gmdb_print_quote(outfile, need_quote, quotechar, delimiter, str);
 			g_value_unset(&value);
 		}
-		fprintf(outfile,lineterm);
+		fprintf(outfile, "%s", lineterm);
 	} while (gtk_tree_model_iter_next(GTK_TREE_MODEL(store), &iter));
 
 	fclose(outfile);
diff -Naurp mdbtools/src/gmdb2/table_export.c mdbtools.oden/src/gmdb2/table_export.c
--- mdbtools/src/gmdb2/table_export.c	2005-01-15 06:02:12.000000000 +0100
+++ mdbtools.oden/src/gmdb2/table_export.c	2008-12-21 20:26:53.000000000 +0100
@@ -142,7 +142,7 @@ gmdb_export_help_cb(GtkWidget *w, gpoint
 
 	gnome_help_display("gmdb.xml", "gmdb-table-export", &error);
 	if (error != NULL) {
-		g_warning (error->message);
+		g_warning ("%s", error->message);
 		g_error_free (error);
 	}
 }
@@ -195,23 +195,23 @@ int rows=0;
 		/* display column titles */
 		col=g_ptr_array_index(table->columns,i);
 		if (need_headers)  {
-			if (i>0) fprintf(outfile,delimiter);
+			if (i>0) fprintf(outfile, "%s", delimiter);
 			gmdb_print_quote(outfile, need_quote, quotechar, delimiter, col->name);
 			fprintf(outfile,"%s", col->name);
 			gmdb_print_quote(outfile, need_quote, quotechar, delimiter, col->name);
 		}
 	}
-	if (need_headers) fprintf(outfile,lineterm);
+	if (need_headers) fprintf(outfile, "%s", lineterm);
 
 	/* fetch those rows! */
 	while(mdb_fetch_row(table)) {
 		for (i=0;i<table->num_cols;i++) {
-			if (i>0) fprintf(outfile,delimiter);
+			if (i>0) fprintf(outfile, "%s", delimiter);
 			gmdb_print_quote(outfile, need_quote, quotechar, delimiter, bound_data[i]);
 			fprintf(outfile,"%s", bound_data[i]);
 			gmdb_print_quote(outfile, need_quote, quotechar, delimiter, bound_data[i]);
 		}
-		fprintf(outfile,lineterm);
+		fprintf(outfile, "%s", lineterm);
 		rows++;
 	}
 
diff -Naurp mdbtools/src/util/mdb-export.c mdbtools.oden/src/util/mdb-export.c
--- mdbtools/src/util/mdb-export.c	2007-03-06 02:16:54.000000000 +0100
+++ mdbtools.oden/src/util/mdb-export.c	2008-12-21 20:26:53.000000000 +0100
@@ -37,7 +37,7 @@ print_col(gchar *col_val, int quote_text
 	gchar *s;
 
 	if (quote_text && is_text_type(col_type)) {
-		fprintf(stdout,quote_char);
+		fprintf(stdout, "%s", quote_char);
 		for (s=col_val;*s;s++) {
 			if (strlen(quote_char)==1 && *s==quote_char[0]) {
 		/* double the char if no escape char passed */
@@ -49,7 +49,7 @@ print_col(gchar *col_val, int quote_text
 			}
 			else fprintf(stdout,"%c",*s);
 		}
-		fprintf(stdout,quote_char);
+		fprintf(stdout, "%s", quote_char);
 	} else {
 		fprintf(stdout,"%s",col_val);
 	}
@@ -177,7 +177,7 @@ main(int argc, char **argv)
 		fprintf(stdout,"%s",sanitize_name(col->name,sanitize));
 		for (j=1;j<table->num_cols;j++) {
 			col=g_ptr_array_index(table->columns,j);
-			fprintf(stdout,delimiter);
+			fprintf(stdout, "%s", delimiter);
 			fprintf(stdout,"%s",sanitize_name(col->name,sanitize));
 		}
 		fprintf(stdout,"\n");
@@ -203,7 +203,7 @@ main(int argc, char **argv)
 				mdb_ole_read(mdb, col, bound_values[j], MDB_BIND_SIZE);
 			}
 			if (j>0) {
-				fprintf(stdout,delimiter);
+				fprintf(stdout, "%s", delimiter);
 			}
 			if (insert_statements && !bound_lens[j]) {
 				print_col("NULL",0,col->col_type, quote_char, escape_char);
@@ -212,7 +212,7 @@ main(int argc, char **argv)
 			}
 		}
 		if (insert_statements) fprintf(stdout,")");
-		fprintf(stdout, row_delimiter);
+		fprintf(stdout, "%s", row_delimiter);
 	}
 	for (j=0;j<table->num_cols;j++) {
 		g_free(bound_values[j]);
